### PR TITLE
NXP-29708 improve rest response for automation + review transaction rollbacks

### PR DESCRIPTION
**Pull request by atchertchian**

### CODE REVIEW

File: modules/nuxeo-apidoc-webengine/src/main/java/org/nuxeo/apidoc/browse/Distribution.java [hunk 1]
```diff
@@ -493,10 +493,13 @@ public Object uploadDistrib() {
                     SUB_DISTRIBUTION_PATH_RESERVED);
         } catch (IOException | IllegalArgumentException | NuxeoException e) {
             log.error(e, e);
-            return getView("importKO").arg("message", e.getMessage()).arg("source", source);
+            return Response.status(Status.INTERNAL_SERVER_ERROR)
+                           .type(MediaType.TEXT_PLAIN)
+                           .entity("Upload not done:" + e.getMessage())
```
Comment (efge) [inline modules/nuxeo-apidoc-webengine/src/main/java/org/nuxeo/apidoc/browse/Distribution.java:538]: Add a space after `:`

File: modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/snapshot/SnapshotManagerComponent.java [hunk 2]
```diff
@@ -107,6 +107,8 @@
 
     protected static final String IMPORT_TMP = "tmpImport";
 
+    protected static final int UPLOAD_SAVE_INTERVAL = 3000;
```
Comment (efge) [inline modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/snapshot/SnapshotManagerComponent.java:110]: Can you add `// save every 3000 documents` to clarify that it's a number of docs and not an interval in seconds or something else

File: modules/nuxeo-apidoc-webengine/src/main/java/org/nuxeo/apidoc/browse/Distribution.java [hunk 3]
```diff
@@ -192,6 +184,13 @@ protected Object show404() {
         return Response.status(404).type(MediaType.TEXT_HTML_TYPE).entity(Resource404.getPageContent()).build();
     }
 
+    protected void commitOrRollbackAndRestartTransaction() {
+        if (TransactionHelper.isTransactionActiveOrMarkedRollback()) {
```
Comment (efge) [inline modules/nuxeo-apidoc-webengine/src/main/java/org/nuxeo/apidoc/browse/Distribution.java:191]: The `if` shouldn't be needed, the caller should always know that it's in a transactional context. (This is a general issue in too much of our code, the no-transaction should be extremely rare and known to the context.)

File: ci/Jenkinsfiles/export.groovy [hunk 4]
```diff
@@ -251,7 +251,7 @@ pipeline {
                     --namespace=${PREVIEW_NAMESPACE}
 
                   ${curlCommand} ${explorerUrl} --output home.html
-                  ${curlCommand} -d 'name=${params.SNAPSHOT_NAME}' -d 'version=${NUXEO_IMAGE_VERSION}' ${explorerUrl}/save
+                  ${curlCommand} -d 'name=${params.SNAPSHOT_NAME}' -d 'version=${NUXEO_IMAGE_VERSION}' -H"Accept: text/plain" ${explorerUrl}/save
```
Comment (efge) [inline ci/Jenkinsfiles/export.groovy:254]: Add a space after `-H` for consistency

File: modules/nuxeo-apidoc-webengine/src/main/java/org/nuxeo/apidoc/browse/Distribution.java [hunk 5]
```diff
@@ -393,26 +421,47 @@ public Object doSaveExtended() {
...
                     readUploadFormData(formData), SUB_DISTRIBUTION_PATH_RESERVED, filter);
-            view = getView("saved").arg("source", source);
+            hasError = false;
         } catch (NuxeoException e) {
             log.error("Error during storage", e);
             TransactionHelper.setTransactionRollbackOnly();
-            view = getView("savedKO").arg("message", e.getMessage()).arg("source", source);
+            hasError = false;
+            errorMessage = e.getMessage();
+
```
Comment (efge) [inline modules/nuxeo-apidoc-webengine/src/main/java/org/nuxeo/apidoc/browse/Distribution.java:446]: Remove extra empty line

File: modules/nuxeo-apidoc-webengine/src/main/java/org/nuxeo/apidoc/browse/Distribution.java [hunk 6]
```diff
@@ -393,26 +421,47 @@ public Object doSaveExtended() {
...
+                view = getView("savedKO").arg("message", errorMessage).arg("source", source);
+            } else {
+                view = getView("saved").arg("source", source);
+            }
+            return view;
+        } else {
+            if (hasError) {
+                return Response.status(Status.INTERNAL_SERVER_ERROR)
+                               .type(MediaType.TEXT_PLAIN)
+                               .entity("Save failed:" + errorMessage)
```
Comment (efge) [inline modules/nuxeo-apidoc-webengine/src/main/java/org/nuxeo/apidoc/browse/Distribution.java:461]: Add space after `:`

**Review by efge — CHANGES_REQUESTED** [review]
Mostly details

**Review by kevinleturc — APPROVED** [review]

**Pull request merged**